src/components/ui/StatusBadge.svelte

```diff
@@ -27,8 +31,14 @@
       <CheckIcon />
     {:else if status === Status.Failed}
       <WarningIcon style="color: {color}" />
+    {:else if status === Status.Canceled}
+      <CloseIcon />
     {:else if status === Status.Incomplete}
-      <SpinnerIcon />
+      {#if determinateProgress}
```

Comment:
Rather than having a variable determine which loading component to show, would it make sense to put the indeterminate version in the `ProgressRadial` component itself? Also, what do you think of making the `SpinnerIcon` show if progress is falsy, instead of having a separate boolean determine which to show? My thinking is that if the progress is 0 for a while, the user essentially doesn't see anything. If they have a spinning indeterminate icon, it could convey that it's at least loading something until it comes back with an actual progress percentage.

Reply:
One issue is that we don't have an indeterminate spinner, just a horizontal determinate progress bar and then a radial determinate progress. Think a while back the animated spinners were nixed since you'd have like 2-3 copies of the spinner spinning at one time..

Reply:
Guess we could have the StatusBadge just show radial progress when the status is incomplete?

Reply:
I see. Yeah I was thinking we could just do a simple rotation animation on the spinner icon or something lol. I'm not sure how useful it is to show the indeterminate state, but just for me, it makes me think something is at least is happening and not frozen.

Reply:
Now there shouldn't be an indeterminate state aside from the "pending/queued" state which is labeled and shouldn't really be a spinner i think?

Reply:
Ah yes. Sorry. I didn't mean "indeterminate" as a potential status state. I meant it specifically for the loading circle component having a determinate/indeterminate mode. 

I think this might just be a me thing cuz I'm used to interactions like deleting a large number of files on Windows where it shows an indeterminate progress bar until it calculates how many files are going to be deleted and then switches to a determinate progress bar. If you think that doesn't make sense in this case, then I'm ok with how it is currently.

Reply:
Oh I follow now, let me adjust some things!

Reply:
Take another look now?